src/app-layer-modbus.c

```diff
@@ -598,6 +598,11 @@ static void ModbusParseReadResponse(ModbusTransaction   *tx,
             goto error;
     }
 
+    tx->data = SCMalloc(count);
```

Comment:
I see multiple places in the code where tx->data is alloc'd, all unconditionally. I worry about the case of overwriting existing allocations and thus memory leaks.